src/versionbits.cpp

```diff
@@ -36,7 +38,8 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
             cache[pindexPrev] = ThresholdState::DEFINED;
             break;
         }
-        if (pindexPrev->GetMedianTimePast() < nTimeStart) {
+        if ((fHeightBased && (pindexPrev->nHeight + 1) < nHeightStart) ||
```

Comment:
Is it cleaner to do a ternary operator here?

`if (fHeightBased ? (pindexPrev->nHeight + 1) < nHeightStart : pindexPrev->GetMedianTimePast() < nTimeStart)`